### PR TITLE
LLM: add UT for starcoder (convert, inference)  update examples and readme

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

add UT for starcoder, update examples and readme.

### 1. Why the change?

add UT for starcoder, update examples and readme.

### 2. User API changes

No change.

### 3. Summary of the change 

- add UT for starcoder(convert, inference)
- update examples and readme

### 4. How to test?
- [x] Unit test

### CODE REVIEW

**Comment by Oscilloscope98** [comment]
Seems that you forgot to update /home/yuwen/BigDL/.github/workflows/llm_unit_tests_linux.yml

**Comment by rnwang04** [comment]
> Seems that you forgot to update /home/yuwen/BigDL/.github/workflows/llm_unit_tests_linux.yml



> Seems that you forgot to update /home/yuwen/BigDL/.github/workflows/llm_unit_tests_linux.yml

Thanks, have fixed it.

File: python/llm/example/transformers/README.md [hunk 1]
```diff
@@ -93,4 +94,14 @@ inference:       total time =     xxxx ms
...
 {'id': 'cmpl-a0ab2953-e08c-449c-b476-e21ad5bb84b0', 'object': 'text_completion', 'created': 1686557434, 'model': './bigdl_llm_bloom_q4_0.bin', 'choices': [{'text': 'Q: What is CPU? A: central processing unit</s>', 'index': 0, 'logprobs': None, 'finish_reason': None}], 'usage': {'prompt_tokens': None, 'completion_tokens': None, 'total_tokens': None}}
+```
+
+### Model family StarCoder
+```log
+bigdl-llm: mem per token =   313912 bytes
+bigdl-llm:     load time =   479.41 ms
+bigdl-llm:   sample time =    34.29 ms
+bigdl-llm:  predict time =  1365.27 ms / 19.50 ms per token
+bigdl-llm:    total time =  1928.46 ms
```
Comment (Oscilloscope98) [inline python/llm/example/transformers/README.md:105]: Please hide the detailed time info as others.
image

And perhaps give an example prediction if it is easy to do :)

**Review by Oscilloscope98 — APPROVED** [review]
Others LGTM

File: python/llm/example/transformers/README.md (hunk 1 above)
Comment (rnwang04) [reply python/llm/example/transformers/README.md]: Done

**Pull request merged**